documentation/dsi-integration.md

```diff
@@ -0,0 +1,171 @@
+
+It replaces the DSI sign in method with one whereby the user is prompted to enter their email address, and clicks a unique login link. This login link works only once and expires after a configurable time.
+
+### How to use
+
+Here are the steps to follow to use our contingency fallback sign-in method.
+
+First, decide whether to switch on the fallback authentication. This call should be made by the Product Owner/Manager if they are available.
+   - An alternative to this fallback sign in method could be replacing DSI with a notice to users telling them that they can't access the service. To do this, we would reinstate the environment variable and code which was deleted in commit [`🔥 Remove FEATURE_SIGN_IN_ALERT flag`](https://github.com/DFE-Digital/teaching-vacancies/commit/bc12fb9808c955f86cd87e62648a76786516e2c3).
+
+### Enabling Authentication Fallback
+
+Switch on the fallback authentication by setting the environment variable `AUTHENTICATION_FALLBACK` to `true`.
+
+The following method is the fastest, but requires `SpaceDeveloper` permission on `production`, and only persists until the next [Automated Deployment to staging and production](/deployments.md#build-and-deploy-to-staging-and-production---github-actions)
+
+- Block all deployments to `production` by requesting in the Slack channel `#tv_engineering`
+- Log in to GOV.UK PaaS (with `cf login --sso`). You will need a [Passcode](https://login.london.cloud.service.gov.uk/passcode)
+- Select the `teaching-vacancies-production` space from the menu, or Switch to the production space:
+```bash
+cf target -s teaching-vacancies-production
+```
+
+- Get the current setting with:
+```bash
+cf env teaching-vacancies-production | grep "AUTHENTICATION_FALLBACK:"
+```
+
+- Update the environment variable with the [set-env](http://cli.cloudfoundry.org/en-US/v7/set-env.html) command:
+```bash
+cf set-env teaching-vacancies-production AUTHENTICATION_FALLBACK true
+```
+
+- [Restage](http://cli.cloudfoundry.org/en-US/v7/restage.html) the app to pick up the updated configuration:
+```bash
+cf restage teaching-vacancies-production --strategy rolling
+```
+
+Changing the environment variable within the `<env>_app_env.yml` is a more persistent change, and allows deployments to production to continue:
+- create a feature branch
+- edit the file [`terraform/workspace-variables/production_app_env.yml`](../terraform/workspace-variables/production_app_env.yml)
+- set the environment variable to
+```
+AUTHENTICATION_FALLBACK: true
+```
+- create a Pull Request
+- merge after approval
+### Disabling Authentication Fallback
+
+Switch off the fallback authentication by setting the environment variable `AUTHENTICATION_FALLBACK` to `false`.
+
+The following method is the fastest, but requires `SpaceDeveloper` permission on `production`, and only persists until the next [Automated Deployment to staging and production](/deployments.md#build-and-deploy-to-staging-and-production---github-actions)
```

Comment:
I think I get why you're repeating-but-not-quite yourself - if the sections were deduplicated, we wouldn't be able to rely on copying and pasting to do the job, which reduces the chance of doing any thinking (and thinking risks making a mistake). However, I dislike it because it makes the document more unwieldy, and because it would be easy to copy and paste from the wrong section of the document if you confused one section with the other. I suggest abstracting out as much as you can of the rapid-flag-switch instructions, and linking to them either in another file or at the end of this one.

Reply:
Have made it much more DRY by using the helpful verb "toggle"